src/formulas/parser.ts

```diff
@@ -241,12 +325,14 @@ export function parse(str: string): AST {
 }
 
 export function parseTokens(tokens: Token[]): AST {
-  tokens = tokens.filter((x) => x.type !== "SPACE");
-  if (tokens[0]?.value === "=") {
-    tokens.splice(0, 1);
+  const richTokens = tokens.map((token, index) => ({ ...token, tokenIndex: index }));
+  const tokensToParse = richTokens.filter((x) => x.type !== "SPACE");
+  if (tokensToParse[0]?.value === "=") {
+    tokensToParse.splice(0, 1);
```

Comment:
instead of `splice`, you could now create the `TokenList` before and `shift` one time